### DISCUSSION TITLE
Test with  [ConditionalFact] Attribute even when false still run VS2022

### DISCUSSION THREAD

**Discussion by PRIMETSS** [xunit/xunit#2433] [Question] [answered]
Running Test in VS2022 seems to run ALL tests even flagged with [ConditionalFact] Attribute which at Debug evaluates to 'false'
![143965196-a4e2468e-7971-40f5-99c9-7863df816916](https://user-images.githubusercontent.com/11331370/144338422-97b573b4-e520-4bd0-9cbf-3265d9958221.png)
Is this expected behaviour?

Trying to work on DOTNetCore (DotNet 6) System.IO.Ports library, but running some tests from CLI hangs for 7+ hours running some tests. But trying to debug them inside VS2022 Test Explorer has isses as it tries to run all tests, even though Condition Attribute is false, and should not run?

Extract from GitHub issues below:
Running tests from CLI `runtime\src\libraries\System.IO.Ports\tests>dotnet build /t:test` after aborting script when it hits a test that runs for >7 hours! `System.IO.Ports.Tests.DiscardOutBuffer.OutBufferFilled_Discard_Multiple()` causes files to be left open (requires reboot to kill, as 'dotnet.exe' or 'xunit.console.exe' cannot be killed via Process Explorer, reboot required to allow any further tests to run without errors.

**Comment by bradwilson** [comment, marked as answer]
Does this only happen in Visual Studio? If so, please try disabling real-time test discovery and let me know if the problem still persists:

<img width="683" alt="image" src="https://user-images.githubusercontent.com/16944/145696366-bec6aa8f-35ed-4a25-b687-7d32709f0c94.png">

**Reply by PRIMETSS** [reply]
Yes in VS2022 and turned off option, closed re-open Solution and still runs
![image](https://user-images.githubusercontent.com/11331370/145700007-6c25d0d7-3323-4a10-9889-9bc7f972f859.png)

From CLI seems OK [SKIP]
![image](https://user-images.githubusercontent.com/11331370/145700141-674ca141-041b-4af2-b3af-cd5ac6532a3f.png)

**** Apologies: Just noticed default branch was [release Net6.0], will check with Main branch!

**Reply by PRIMETSS** [reply]
NET7.0 (main) 

Hmmm in main I can run tests on System.IO.Ports from CLI and it correctly [SKIP]s tests "HasNullModem" condition
`build /t:Test`  From System.IO.Ports.Tests

(however test `System.IO.Ports.Tests.DiscardOutBuffer.OutBufferFilled_Discard_Once` runs > 7hrs, which is what I want to look at why, happened in Net6 release too)

VS2022 Test Explorer ERRORS because System.IO.Ports.native .dll is not in artifacts, ie its never built...
But no matter what I build or try (even `build-native.cmd` in .\src\native\libs) cannot get System.IO.Ports.Native.dll built!

Tried full builds, clr, libs, libs.tests, libs.pretests never get System.IO.Ports.native.dll generated!???
Fetch main, clean bin, then full build Libs as Debug

Don't know how to get that built so can try in VS2022 Testing....

![image](https://user-images.githubusercontent.com/11331370/145743911-a41bb90b-0792-4e62-ae37-5d5be33f980c.png)

**Reply by bradwilson** [reply]
This seems to be a question that's better answered by the .NET team, since these tests (and `ConditionalFact`) are code that was written and owned by them.
